starclass/todolist.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+
+.. codeauthor:: Rasmus Handberg <rasmush@phys.au.dk>
+"""
+
+import os
+import re
+import sqlite3
+from contextlib import closing
+from .io import load_lightcurve
+
+#--------------------------------------------------------------------------------------------------
+def create_fake_todolist(input_folder, output_todo='todo.sqlite', file_pattern=None,
+	overwrite=False):
+	"""
+
+	.. codeauthor:: Rasmus Handberg <rasmush@phys.au.dk>
+	"""
+
+	# Basic checks of the input:
+	if not os.path.isdir(input_folder):
+		raise NotADirectoryError(input_folder)
+	if file_pattern is None:
+		file_pattern = r'\.fits(\.gz)?$'
+
+	# Go through the input directory recursively and find all files which match the pattern:
+	files = []
+	regex = re.compile(file_pattern)
+	for root, dirnames, filenames in os.walk(input_folder, followlinks=True):
+		for filename in filenames:
+			if regex.match(filename):
+				files.append(os.path.join(root, filename))
+
+	if not files:
+		raise ValueError("No files were found")
+
+	# Check path to the TODO-file, and delete if it already exists:
+	todo_file = os.path.join(input_folder, output_todo)
+	if os.path.exists(todo_file):
+		if overwrite:
+			os.remove(todo_file)
+		else:
+			raise ValueError()
+
+	# Open the todo-file and create the records of the files in it:
+	try:
+		with closing(sqlite3.connect(todo_file)) as conn:
+			cursor = conn.cursor()
+
+			todolist_structure(conn)
+
+			for k, fpath in enumerate(files):
+
+				lightcurve = os.path.relpath(fpath, input_folder).replace('\\', '/')
+
+				lc = load_lightcurve(fpath)
+				starid = lc.targetid
+
+				todolist_insert(cursor,
+					priority=k+1,
+					starid=starid,
+					lightcurve=lightcurve)
+
```

Comment:
add the following two lines right after the for loop (otherwise the todo stays empty):
`conn.commit()`
`cursor.close()`

Reply:
Fixed in latest version